### PR TITLE
Milestone-1: Add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by carlylechia**
To fulfill this task, I:

- [x] Created the `class Person` with the following:

   - Instance vars: @id, @name, and @age.

   - Constructor with name, age, and parent_permission as parameters. name and parent_permission are optional and have default values of "Unknown" and true.

   - Getters for @id, @name, and @age.

   - Setters for @name and @age.

   - Private method is_of_age? which returns true if @age is greater or equal to 18 and false otherwise.

   - The public method can_use_services? returns true if a person is of age or has permission from parents.

- [x] Created the `class Student` with the following:

   - Inherits from Person.

   - Constructor extends parent's constructor by adding @classroom and a parameter for it.

   - Method play_hooky that returns "¯\(ツ)/¯".

- [x] Created the `class Teacher` with the following:

   - Inherits from Person.

   - Constructor extends parent's constructor by adding @specialization and a parameter for it.

   - Override can_use_services? so it always returns true.

> Each class was saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    @id = Random.rand(1..1000)
+  end
+
+  def can_use_services?
+    return true if of_age? || parent_permission == true
+
+    false
+  end
+
+  def of_age?
```
Comment (youmari) [inline person.rb:18]: - [ ] This function should be `is_of_age?` not `of_age?` please try to fix this typo

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @carlylechia  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great title and short summary 
✔️ Git flow 
✔️ No linter error

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (carlylechia) [reply person.rb:18]: This name throws an error. It's against rubocop linters rules

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
## Hello @carlylechia👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by rustam-tolipov — APPROVED** [review]
## Hello @carlylechia 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**